src/reporting/clients/IAggregationsClient.ts

```diff
@@ -0,0 +1,278 @@
+*--------------------------------------------------------------------------------------------*/
+import type { AccessToken } from "@itwin/core-bentley";
+import { AggregationProperty, AggregationPropertyCreate, AggregationPropertyUpdate, AggregationTable, AggregationTableCreate, AggregationTableSet, AggregationTableSetCreate, AggregationTableSetUpdate, AggregationTableUpdate } from "../interfaces/AggregationProperties";
+import type { EntityListIterator } from "../../common/iterators/EntityListIterator";
+
+export interface IAggregationsClient {
+  /**
+   * Gets all AggregationProperties in a given aggregation table. This method returns the full list of Aggregation Properties.
+   * @param {string} accessToken OAuth access token with scope `insights:read`.
+   * @param {string} aggregationTableSetId Id of the aggregation table set.
+   * @param {string} aggregationTableId Id of the aggregation table.
+   * @param {number} top The number of entities to load per page.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-aggregationproperties/
+   */
+  getAggregationProperties(
+    accessToken: AccessToken,
+    aggregationTableSetId: string,
+    aggregationTableId: string,
+    top?: number
+  ): Promise<AggregationProperty[]>;
+
+  /**
+   * Gets an async paged iterator of AggregationProperties in a given aggregation table.
+   * This method returns an iterator which loads pages of Aggregation Properties as it is being iterated over.
+   * @param {string} accessToken OAuth access token with scope `insights:read`.
+   * @param {string} aggregationTableSetId Id of the aggregation table set.
+   * @param {string} aggregationTableId Id of the aggregation table.
+   * @param {number} top The number of entities to load per page.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-aggregationproperties/
+   */
+  getAggregationPropertiesIterator(
+    accessToken: AccessToken,
+    aggregationTableSetId: string,
+    aggregationTableId: string,
+    top?: number
+  ): EntityListIterator<AggregationProperty>;
+
+  /**
+   * Retrieves an aggregation property.
+   * @param {string} accessToken OAuth access token with scope `insights:read`.
+   * @param {string} aggregationTableSetId Id of the aggregation table set.
+   * @param {string} aggregationTableId Id of the aggregation table.
+   * @param {string} aggregationPropertyId Id of the aggregation property.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-aggregationproperty/
+   */
+  getAggregationProperty(
+    accessToken: AccessToken,
+    aggregationTableSetId: string,
+    aggregationTableId: string,
+    aggregationPropertyId: string,
+  ): Promise<AggregationProperty>;
+
+  /**
+   * Updates an aggregation property.
+   * @param {string} accessToken OAuth access token with scope `insights:read`.
```

Comment:
Wrong scope.

Reply:
Review all operations and check if the scopes in comments match the API documentation in developer.bentley.com.